plugins/janus_videoroom.c

```diff
@@ -2024,16 +2059,21 @@ void janus_videoroom_incoming_data(janus_plugin_session *handle, char *buf, int
 	janus_videoroom_session *session = (janus_videoroom_session *)handle->plugin_handle;
 	if(!session || g_atomic_int_get(&session->destroyed) || session->participant_type != janus_videoroom_p_type_publisher)
 		return;
-	janus_videoroom_publisher *participant = (janus_videoroom_publisher *)session->participant;
-	if(participant == NULL || g_atomic_int_get(&participant->destroyed))
+	janus_videoroom_publisher *participant = janus_videoroom_session_get_publisher(session);
+	if(participant == NULL || g_atomic_int_get(&participant->destroyed)) {
+		g_clear_pointer(&participant, janus_videoroom_publisher_dereference);
```

Comment:
Yep, but we may be doing an `&NULL`. I assume this is safe then? (probably a dumb question :smile: )